server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CassandraDumbBlobStoreCache.java

```diff
@@ -0,0 +1,119 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+import javax.inject.Inject;
+
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor;
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.cassandra.BlobTables;
+import org.apache.james.blob.cassandra.utils.DataChunker;
+import org.reactivestreams.Publisher;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.google.common.base.Preconditions;
+
+import reactor.core.publisher.Mono;
+
+public class CassandraDumbBlobStoreCache implements DumbBlobStoreCache {
+
+    private final DataChunker dataChunker;
+    private final CassandraAsyncExecutor cassandraAsyncExecutor;
+    private final PreparedStatement insert;
+    private final PreparedStatement select;
+    private final PreparedStatement delete;
+    private final CassandraCacheConfiguration cacheConfiguration;
+
+    @Inject
+    public CassandraDumbBlobStoreCache(Session session, CassandraCacheConfiguration cacheConfiguration) {
+        this.dataChunker = new DataChunker();
+        this.cassandraAsyncExecutor = new CassandraAsyncExecutor(session);
+        this.insert = prepareInsert(session);
+        this.select = prepareSelect(session);
+        this.delete = prepareDelete(session);
+        this.cacheConfiguration = cacheConfiguration;
+    }
+
+    private PreparedStatement prepareDelete(Session session) {
+        return session.prepare(
+            QueryBuilder.delete().
+                from(BlobTables.BucketBlobTable.TABLE_NAME)
+                .where(eq(ID, bindMarker(ID))));
+    }
+
+    private PreparedStatement prepareSelect(Session session) {
+        return session.prepare(
+            QueryBuilder.select()
+                .from(BlobTables.BucketBlobTable.TABLE_NAME)
+                .where(eq(ID, bindMarker(ID))));
+    }
+
+    private PreparedStatement prepareInsert(Session session) {
+        return session.prepare(
+            QueryBuilder.insertInto(BlobTables.DumbBlobCache.TABLE_NAME)
+                .value(ID, bindMarker(ID))
+                .value(DATA, bindMarker(DATA)));
+    }
+
+    private boolean isAValidStream(InputStream bufferedData) throws IOException {
+        bufferedData.mark(0);
+        bufferedData.skip(cacheConfiguration.MAX_SIZE);
+        boolean isItABigStream = bufferedData.read() != -1;
+        bufferedData.reset();
+        return isItABigStream;
+    }
+
+    @Override
+    public Publisher<Void> cache(BlobId blobId, byte[] data) {
+        Preconditions.checkNotNull(data, "Data must not be null");
+        Preconditions.checkArgument(data.length <= 0, "Data must not be empty");
+        Preconditions.checkArgument(data.length > cacheConfiguration.MAX_SIZE, "Data must not biger than %s kb", cacheConfiguration.MAX_SIZE);
```

Comment:
If data is bigger than the cache size then we do not save it. We return Mono.empty().

Reply:
ok